### PR TITLE
A0-2814: port metrics from release-11

### PR DESCRIPTION
**Pull request by kostekIV**
# Description

Port metrics from the release-11 branch and include new ones.

## Type of change

Please delete options that are not relevant.
- Breaking change - breaking as part of the grafana dashboard will stop working if we do not update

### CODE REVIEW

File: finality-aleph/src/sync/service.rs [hunk 1]
```diff
@@ -265,6 +284,7 @@ where
             peer,
             response_items,
         );
+        self.metrics.report_sync_handle_request_response();
```
Comment (lesniak43) [inline finality-aleph/src/sync/service.rs:287]: to match below

File: finality-aleph/src/sync/service.rs [hunk 2]
```diff
@@ -328,6 +350,7 @@ where
 
     fn handle_task(&mut self, task: RequestTask<BlockIdFor<J>>) {
         trace!(target: LOG_TARGET, "Handling task {}.", task);
+        self.metrics.report_sync_handle_task();
```
Comment (lesniak43) [inline finality-aleph/src/sync/service.rs:353]: this is fine I think, but maybe move to the very end for clarity?

**Review by lesniak43 — APPROVED** [review]

File: finality-aleph/src/sync/service.rs (hunk 1 above)
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs:287]: this is also trickier than it looks

File: finality-aleph/src/sync/service.rs [hunk 3]
```diff
@@ -164,13 +172,21 @@ where
         };
         let (request, peers) = pre_request.with_state(state);
         trace!(target: LOG_TARGET, "Sending a request: {:?}", request);
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs:175]: I cant do that without discardin request information that could go into trace log

File: finality-aleph/src/sync/service.rs [hunk 4]
```diff
@@ -145,6 +152,7 @@ where
             }
         };
         trace!(target: LOG_TARGET, "Broadcasting state: {:?}", state);
+        self.metrics.report_sync_broadcast();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: Done (without moving trace to the top)

File: finality-aleph/src/sync/service.rs [hunk 5]
```diff
@@ -164,13 +172,21 @@ where
         };
         let (request, peers) = pre_request.with_state(state);
         trace!(target: LOG_TARGET, "Sending a request: {:?}", request);
+        self.metrics.report_sync_send_request();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 6]
```diff
@@ -164,13 +172,21 @@ where
...
     }
 
     fn send_to(&mut self, data: NetworkData<B, J>, peer: N::PeerId) {
+        trace!(
+            target: LOG_TARGET,
+            "Sending data {:?} to peer {:?}",
+            data,
+            peer
+        );
+        self.metrics.report_sync_send_to();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 7]
```diff
@@ -184,6 +200,7 @@ where
             state,
             peer
         );
+        self.metrics.report_sync_handle_state();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 8]
```diff
@@ -242,6 +260,7 @@ where
             "Handling a justification {:?} from user.",
             justification,
         );
+        self.metrics.report_sync_handle_justification_from_user();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 9]
```diff
@@ -292,6 +312,8 @@ where
             request,
             peer
         );
+        self.metrics.report_sync_handle_request();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs:322]: done

File: finality-aleph/src/sync/service.rs (hunk 2 above)
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 10]
```diff
@@ -341,6 +364,7 @@ where
         match event {
             BlockImported(header) => {
                 trace!(target: LOG_TARGET, "Handling a new imported block.");
+                self.metrics.report_sync_handle_block_imported();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

File: finality-aleph/src/sync/service.rs [hunk 11]
```diff
@@ -363,6 +388,7 @@ where
             "Handling an internal request for block {:?}.",
             id,
         );
+        self.metrics.report_sync_handle_internal_request();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: done

**Pull request merged**